api/client.md

```diff
@@ -0,0 +1,646 @@
+---
+description: >-
+  The Client struct represents the HTTP Client. It has methods like Get, Post, etc, they'll create a new Agent instance and parse request url automatically.
+---
+
+# ⚡ Client
+
+## Start request
+Start a http request with http method and url.
+
+{% code title="Signatures" %}
+```go
+// Client http methods
+func (c *Client) Get(url string) *Agent
+func (c *Client) Head(url string) *Agent
+func (c *Client) Post(url string) *Agent
+func (c *Client) Put(url string) *Agent
+func (c *Client) Patch(url string) *Agent
+func (c *Client) Delete(url string) *Agent
+```
+{% endcode %}
+
+## ✨ Agent
+`Agent` is on top of [fasthttp `HostClient`](https://github.com/valyala/fasthttp/blob/master/client.go#L603) which has lots of convenient helper methods and several `send request` methods.
+
+### Parse
+Parse initializes URI and HostClient.
+
+{% code title="Parse" %}
+```go
+a := AcquireAgent()
+req := a.Request()
+req.Header.SetMethod(MethodGet)
+req.SetRequestURI("http://example.com")
+
+if err := a.Parse(); err != nil {
+	panic(err)
+}
+
+code, body, errs := a.Bytes() // ...
+```
+{% endcode %}
+
+### Set
+Set sets the given 'key: value' header.
+
+{% code title="Signature" %}
+```go
+func (a *Agent) Set(k, v string) *Agent
+func (a *Agent) SetBytesK(k []byte, v string) *Agent
+func (a *Agent) SetBytesV(k string, v []byte) *Agent
+func (a *Agent) SetBytesKV(k []byte, v []byte) *Agent
+```
+{% endcode %}
+
+{% code title="Example" %}
+```go
+
+agent.Set("k1", "v1").
+    SetBytesK([]byte("k1"), "v1").
+    SetBytesV("k1", []byte("v1")).
+    SetBytesKV([]byte("k2"), []byte("v2"))
+// ...
+```
+{% endcode %}
+
+### Add
+Add adds the given 'key: value' header. Multiple headers with the same key may be added with this function.
```

Comment:
> Multiple headers with the same key may be added with this function.
So does this mean the header's won't overwrite eachother?

Reply:
Yeah you are right. we can have headers like
```
K1: v1
K1: v2
```